File: solutions/day_2/valid_anagram.dart
class Solution {
  bool isAnagram(String s, String t) {
    if (s.length != t.length) {
      return false;
    }
// double dot operator is used to chain methods together
// double dot is used for the sort method so as not to return a new list or void error
    List<String> sList = s.split('')..sort();
    List<String> tList = t.split('')..sort();

    for (var i = 0; i < sList.length; i++) {
      if (sList[i] != tList[i]) {
        return false;
      }
    }
    return true;
  }
}

File: validAnagram/validAnagram.js
var isAnagram = function(s, t) {
    sA = s.split('').sort()
    tA = t.split('').sort()

    if(sA.length != tA.length){
        return false
    }
    for(let i=0; i<sA.length; i++)
  
    if(sA[i] != tA[i]){
        return false
    }
    return true
};